front/src/applications/operationalStudies/components/Scenario/AddOrEditScenarioModal.tsx

```diff
@@ -105,48 +119,58 @@ export default function AddOrEditScenarioModal({
   const createScenario = async () => {
     if (!currentScenario.name || !currentScenario.infra_id) {
       setDisplayErrors(true);
-    } else {
-      try {
-        const result = await post(`${rootURI}`, { ...currentScenario, infra_id: infraID });
-        dispatch(updateScenarioID(result.id));
-        navigate('/operational-studies/scenario');
-        closeModal();
-      } catch (error) {
-        console.error(error);
-      }
+    } else if (infraID && projectID && studyID && currentScenario && infraID) {
+      postScenario({
+        projectId: projectID,
+        studyId: studyID,
+        scenarioRequest: { ...(currentScenario as ScenarioRequest), infra_id: infraID },
+      })
+        .unwrap()
+        .then(({ id }) => {
+          dispatch(updateScenarioID(id));
+          navigate('/operational-studies/scenario');
+          closeModal();
+        })
+
+        .catch((error) => console.error(error));
     }
   };
 
   const updateScenario = async () => {
     if (!currentScenario.name) {
       setDisplayErrors(true);
-    } else if (scenario) {
-      try {
-        await patch(`${rootURI}${scenario.id}/`, currentScenario);
-        if (getScenarioTimetable) getScenarioTimetable(true);
-        closeModal();
-      } catch (error) {
-        console.error(error);
-      }
+    } else if (scenario && projectID && studyID && scenario.id) {
+      patchScenario({
+        projectId: projectID,
+        studyId: studyID,
+        scenarioId: scenario.id,
+        scenarioPatchRequest: currentScenario,
+      })
+        .unwrap()
+        .then(() => {
+          if (getScenarioTimetable) getScenarioTimetable(true);
```

Comment:
We should use tags here : if we patch the scenario, then it should unvalidate automatically the scenario and the timetable. Use RTK tags to do this